full width images + fixed nav menu overlap issue

darkpivot opened the pull request:
My goal was to get the images to display like they did on cohost, so they now take up the full width edge-to-edge. This required moving the padding styles to just the text portion of a post. Also fixed an issue with the side nav menu overlapping the top header bar.

darkpivot commented:
This is my first pull request I've ever done on something before, so forgive me if I miss something. Github is saying that it failed a check because I don't have permission to edit the repo, not entirely sure what that's about.

vivlim commented:
Thanks!

The failed check is one I set up to build and deploy to a temporary url, it requires manual approval because it can potentially mess with the repo. I'll run it and try out the change.

vivlim commented on src/views/htmlsandbox.tsx:
```diff
@@ -36,7 +36,7 @@ const HtmlSandbox: Component<HtmlSandboxProps> = (props) => {
             <Match when={sanitizedHtml.state === "ready"}>
                 <div
-                    class="post-content overflow-auto break-words"
+                    class="post-content p-3 overflow-auto break-words"
```
Instead of defining padding on `HtmlSandbox` itself, which is used in multiple places, I would prefer to have each place that uses `HtmlSandbox` be able to optionally set padding (and maybe other classes) on the `HtmlSandbox` component.

I think probably the cleanest way to accomplish that is to set up something similar to how `splitProps` is used in `PostBody` and `CardContent`.

darkpivot replied:
still learning the ropes of solidjs, so I don't understand props much just yet. i ended up adding a wrapper around the htmlsandbox element in the postbody where I could add the padding in the meantime, hopefully that's alright for now.

vivlim commented on src/views/facetnavigation.tsx:
```diff
@@ -85,7 +85,6 @@ export const FacetNavigationFrame: Component<{ children: JSX.Element }> = (
                     "text-card-foreground": true,
                     "shadow-sm": true,
                     fixed: true,
-                    "z-40": true,
```
Thanks! I think at one point this was needed to make the nav menu in mobile view appear above the main content, but that doesn't seem necessary anymore.

vivlim commented on src/components/ui/card.tsx:
```diff
@@ -32,7 +32,7 @@ const CardDescription: Component<ComponentProps<"p">> = (props) => {
 const CardContent: Component<ComponentProps<"div">> = (props) => {
   const [local, others] = splitProps(props, ["class"])
-  return <div class={cn("p-6 pt-0", local.class)} {...others} />
+  return <div class={cn(local.class)} {...others} />
```
Looks good to me, but one consequence to removing this is that the about page (`notsignedin.tsx`) and login page (`login.tsx`) are now both missing padding. Could you please add some back? It doesn't need to be the exact same amount as before, just more than 0.

darkpivot replied:
added padding back to both pages.

vivlim reviewed (changes requested):
Overall looks good, there are some seemingly unintentional side effects because some of the components are used in multiple locations but nothing major.

Congrats on opening your first pr!

darkpivot commented:
Okay, thank you for the feedback! I'll address these later when I get some more time. Apologies for the conflicts, still getting a feel for how everything is laid out.

vivlim commented:
No problem. Let me know if you have any questions I can help with :)

darkpivot commented:
whoops, i think I messed up committing the changes instead of applying them to the suggestions, still learning 😅

i just incorporated the feedback and also fixed up some padding that went missing on the notsignedin page as well

vivlim reviewed (approved):
Looks good, thanks for making the changes!

darkpivot commented:
@vivlim i have a question. since there's a conflict, what's a good way to handle this? i tried resolving it manually by putting my changes back into the most recent version of main, but i'm not 100% sure that it's correct and won't break anything. should I commit it to a new branch? or will committing it to main be ok? i haven't done this before so I just want to make sure I don't break anything 😅

vivlim commented:
hm. I think probably the best thing to do would be to rebase your branch on top of my repo's main branch, then force push the resulting main branch to your repo. before doing the rebase you can create another branch just to have a backup of your changes. I'm not sure what tool you're using for git or what your level of comfort with it is, so if you have questions please lmk.

darkpivot commented:
okay, that makes sense! I think what I did here worked for now since it was a very small change, but I will keep that in mind for the future.

vivlim commented:
I rebased & resolved the conflicts, and pushed them to main; these changes are now in.
